src/constructs/dns/dns-records.ts

```diff
@@ -0,0 +1,57 @@
+import { CfnResource } from "@aws-cdk/core";
+import type { GuStack } from "../core";
+
+export enum RecordType {
+  CNAME = "CNAME",
+}
+
+export interface GuDnsRecordSetProps {
+  name: string;
+  recordType: RecordType;
+  resourceRecords: string[];
+  ttl?: number;
+}
+
+/**
+ * Do not use this construct directly. Use [[`GuCname`]] instead in order to reduce boilerplate.
+ */
+export class GuDnsRecordSet {
+  constructor(scope: GuStack, id: string, props: GuDnsRecordSetProps) {
+    // The spec for this private resource type can be found here:
+    // https://github.com/guardian/cfn-private-resource-types/tree/main/dns/guardian-dns-record-set-type/docs#syntax
+    new CfnResource(scope, id, {
+      type: "Guardian::DNS::RecordSet",
+      properties: {
+        Name: props.name,
+        ResourceRecords: props.resourceRecords,
+        RecordType: props.recordType,
+        TTL: props.ttl ?? 3600,
```

Comment:
This seemed like a sensible default to me but other opinions are welcome.

Reply:
I am not convinced that we should have a default. Perhaps one option might be to provide some presets with purposes?

Reply:
Some kind of presets/guidance would certainly be helpful. I would guess that most users (and I include myself in this group!) won't know what a sensible choice is here.

Do you think it makes sense to have defaults for different record types (i.e. provide a default for CNAMEs) or is that still unhelpful?

Reply:
NS records should certainly be longer. But I'm not entirely convinced that there will be a use case for that with a single plane of control. It's generally about purpose. We could have a default of 3600 on the GuCname type. Think I'd be somewhat happier with that, but again we'd likely want to reduce it during migrations.

Reply:
I wonder if we provide an enum or similar:

```typescript
const TTLs = {
  InMigrationMode: 60,
  StableApplication: 3600
}
```

The intention being to stop you having to think what a sensible value should be. (Obvs. better names needed!)

Reply:
> We could have a default of 3600 on the GuCname type. Think I'd be somewhat happier with that, but again we'd likely want to reduce it during migrations.

I've gone for applying a default at `GuCname` level for now. I agree that we will need to drop this for migrations, but in order to adopt the record in the first place users will need to match the TTL that's currently configured in NS1 (and hopefully this will frequently align with the 1 hour default!)